### PR TITLE
feat(cloudsql): Add mysql and postgres samples for enterprise_plus edition

### PR DESCRIPTION
**Pull request by ibhaskar2**
## Description
Added samples for  cloud sql enterprise plus instance.

Fixes #<ISSUE-NUMBER>

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/sql/docs/postgres/create-instance
                           https://cloud.google.com/sql/docs/postgres/clone-instance
                           https://cloud.google.com/sql/docs/postgres/label-instance

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by ibhaskar2** [comment]
@msampathkumar can you review it? Thanks

File: cloud_sql/postgres_enterprise_plus_instance_labels/main.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START cloud_sql_postgres_enterprise_plus_instance_labels]
+resource "google_sql_database_instance" "postgres_enterprise_plus_instance_labels" {
```
Comment (msampathkumar) [inline cloud_sql/postgres_enterprise_plus_instance_labels/main.tf:18]:
```suggestion
resource "google_sql_database_instance" "default" {
```

File: cloud_sql/postgres_enterprise_plus_instance_pvp/main.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START cloud_sql_postgres_enterprise_plus_instance_pvp]
+resource "google_sql_database_instance" "postgres_pvp_enterprise_plus_instance_name" {
```
Comment (msampathkumar) [inline cloud_sql/postgres_enterprise_plus_instance_pvp/main.tf:18]:
```suggestion
resource "google_sql_database_instance" "default" {
```

**Comment by msampathkumar** [comment]
Overall, lgtm. Lets wait for the CICD to finish in green.

**Review by msampathkumar — COMMENTED** [review]
Nice work @ibhaskar2 Filenames are in-order and naming structure look good. Lets wait for CICD.

**Comment by msampathkumar** [comment]
CICD recorded error like below. Said that `sqlserver_instance_private_ip` error seems un-related to this...


TestSamples/0/sqlserver_instance_private_ip 2023-11-29T16:11:08Z command.go:185: Error: Unable to remove Service Networking Connection, err: Error waiting for Delete Service Networking Connection: Error code 9, message: Failed to delete connection; Producer services (e.g. CloudSQL, Cloud Memstore, etc.) are still using this connection.

**Comment by msampathkumar** [comment]
CICD failed again with following error. Issues needs to be fixed. Kindly check with CodeOwners team for proper resolution.

```
Error: Error, failed to create instance sqlserver-instance: googleapi: Error 400: Invalid request: PSC connectivity can not be enabled for database type SQLSERVER_2019_ENTERPRISE.., invalid
```

**Comment by glasnt** [comment]
I'm not sure if the tests that are being run here are all the cloud_sql/* tests, rather than just the new sub-folder ones. 

Previous error of `Error 400: Invalid request: PSC connectivity can not be enabled for database type SQLSERVER_2019_ENTERPRISE` might be related to https://github.com/hashicorp/terraform-provider-google/issues/16257, which may require project-level attention, if there's org policies in play restricting this.

**Comment by glasnt** [comment]
I've added a skiptest for sqlserver_instance_psc so that CI passes, as it's failing unrelated tests. 

/gcbrun

**Comment by ibhaskar2** [comment]
@glasnt , CICD still shows "Needs /gcbrun from a collaborator"

**Comment by glasnt** [comment]
/gcbrun

My comment was _before_ the commit, which is why it didn't fire 🫠

**Comment by ibhaskar2** [comment]
Hi @glasnt , cicd failed with following error 
`TestSamples/0/sqlserver_instance_private_ip 2024-01-24T06:50:19Z command.go:185: Error: Unable to remove Service Networking Connection, err: Error waiting for Delete Service Networking Connection: Error code 9, message: Failed to delete connection; Producer services (e.g. CloudSQL, Cloud Memstore, etc.) are still using this connection.` 

Is it possible this is cause by my change, if yes can you help me fix it? Thanks

**Comment by glasnt** [comment]
This isn't caused by your change. I've set that tests to be skipped, like earlier. This should make sure your change is merged once CI completes.

**Review by glasnt — APPROVED** [review]

**Pull request merged**